rest-server/src/templates/dockerContainerScript.mustache

```diff
@@ -204,14 +204,49 @@ else
       "[INFO]" "begin to download ssh key pair from hdfs ..."
   hdfs dfs -get "${hdfs_ssh_folder}/.ssh/" "/root/" || exit 1
 fi
-# Start ssh service
-start_ssh_service
+chmod 400 ~/.ssh/$APP_ID
 # Generate ssh connect info file in "PAI_CONTAINER_ID-PAI_CURRENT_CONTAINER_IP-PAI_CONTAINER_SSH_PORT" format on hdfs
 hdfs dfs -touchz ${hdfs_ssh_folder}/$PAI_CONTAINER_ID-$PAI_CONTAINER_HOST_IP-$PAI_CONTAINER_SSH_PORT || exit 1
 
+# Generate ssh config
+ssh_config_path=${HDFS_LAUNCHER_PREFIX}/${PAI_USER_NAME}/${PAI_JOB_NAME}/ssh/config
+hdfs dfs -mkdir -p ${ssh_config_path} || exit 1
+hdfs dfs -touchz ${ssh_config_path}/$APP_ID-$PAI_CURRENT_TASK_ROLE_NAME-$PAI_CURRENT_TASK_ROLE_CURRENT_TASK_INDEX-$PAI_CONTAINER_HOST_IP-$PAI_CONTAINER_SSH_PORT || exit 1
```

Comment:
What if `$PAI_CURRENT_TASK_ROLE_NAME` contains a dash (`'-'`)?

Reply:
pls refer to the `taskRole.name` schema in [job tutorial](https://github.com/Microsoft/pai/tree/master/job-tutorial#json-config-file-for-job-submission)

Reply:
That's a problem, I will fix it.

Reply:
do we have a checker in rest-server to check whether the name complies with the regex?

---
In reply to: [175660963](https://github.com/Microsoft/pai/pull/390#discussion_r175660963) [](ancestors = 175660963)

Reply:
yes, but the regex is `^[A-Za-z0-9\-._~]+$` which supports `-`